examples/glue_benchmark/glue.py

```diff
@@ -158,26 +175,39 @@ def generate_submission_files(finetuning_model, test_ds):
     if not os.path.exists(FLAGS.submission_directory):
         os.makedirs(FLAGS.submission_directory)
     filename = FLAGS.submission_directory + "/" + filenames[FLAGS.task_name]
-
     labelname = labelnames.get(FLAGS.task_name)
+
+    predictions = finetuning_model.predict(test_ds)
+    if FLAGS.task_name == "stsb":
+        predictions = np.squeeze(predictions)
+    else:
+        predictions = np.argmax(predictions, -1)
+
+    # Map the predictions to the right index order.
+    contents = ["" for _ in idx_order]
+    for i, pred in enumerate(predictions):
+        if labelname:
+            pred_value = labelname[int(pred)]
+        else:
+            pred_value = pred
+            if FLAGS.task_name == "stsb":
+                pred_value = min(pred_value, 5)
+                pred_value = max(pred_value, 0)
+                pred_value = f"{pred_value:.3f}"
+        contents[idx_order[i]] = pred_value
+
     with tf.io.gfile.GFile(filename, "w") as f:
         # GLUE requires a format of index + tab + prediction.
         writer = csv.writer(f, delimiter="\t")
         # Write the required headline for GLUE.
         writer.writerow(["index", "prediction"])
-        predictions = finetuning_model.predict(test_ds)
-        predictions = np.argmax(predictions, -1)
-        for idx, pred in enumerate(predictions):
-            if labelname:
-                pred_value = labelname[int(pred)]
-            else:
-                pred_value = pred
-            writer.writerow([idx, pred_value])
 
+        for idx, value in enumerate(contents):
+            writer.writerow([idx, value])
 
-def main(_):
-    train_ds, test_ds, val_ds = load_data(FLAGS.task_name)
 
+def main(_):
+    train_ds, test_ds, val_ds, idx_order = load_data(FLAGS.task_name)
     # ----- Custom code block starts -----
     bert_model = keras_nlp.models.Bert.from_preset("bert_tiny_uncased_en")
```

Comment:
I think this line is only needed if `load_finetuning_model == False` right? Why not move this line into the other custom code block...

```
# ----- Custom model block starts -----
# Other comments....
backbone_model = keras_nlp.models.Bert.from_preset("bert_tiny_uncased_en")
finetuning_model = keras_nlp.models.BertClassifier(
    backbone=backbone_model,
    num_classes=num_classes,
)
# ----- Custom model block ends -----
```

Reply:
We still need the `test_ds` for evaluation, so I am just grouping all of the loading together. It should not be time-consuming, because the user should expect to train at least once before loading a finetuned model.

Reply:
But nothing will use `bert_model` right? It just becomes a dangling loaded model that will eat into your memory. If this script is updated to work with larger model checkpoints, that could be a real issue!

We should not spuriously load something that could be fairly large into memory when we will completely bypass it and load anther replica of the same thing in the `keras.models.load_model()` branch. If we want this to be useful for real work of testing models, we have to "do the right thing" efficiency-wise here.

Reply:
Oh I thought you mean the dataset. 

The model should be moved.